probe-rs/src/core.rs

```diff
@@ -163,6 +174,244 @@ pub trait CoreInterface: MemoryInterface {
     }
 }
 
+/// A snapshot representation of a core state.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CoreDump {
+    /// The registers we dumped from the core.
+    pub registers: HashMap<RegisterId, RegisterValue>,
```

Comment:
I dont think we should store more than necessary and storing the entire Debug register was not possible due to derives/trait requirements.

Reply:
I'm fine with only storing this, but then it should be translated back to `DebugRegisters` when it's read from the file.

Reply:
What's the benefit of that? That will require extra shenanigans :)

Reply:
It is the format in which it is used by all the debugger code, so it will end up in that format anyways. But if this format is present here already, we don't the additional `CoreInterface`, but can just use the registers and `MemoryInterface` for unwinding.